src/ReverseProxy/Routing/DirectForwardingIEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,81 @@
+
+/// <summary>
+/// Extension methods for <see cref="IEndpointRouteBuilder"/> used to add direct forwarding to the ASP.NET Core request pipeline.
+/// </summary>
+public static class DirectForwardingIEndpointRouteBuilderExtensions
+{
+    private static HttpMessageInvoker? _defaultTransport;
+
+    private static HttpMessageInvoker DefaultTransport
+    {
+        get
+        {
+            _defaultTransport ??= new HttpMessageInvoker(new SocketsHttpHandler()
+            {
+                UseProxy = false,
+                AllowAutoRedirect = false,
+                AutomaticDecompression = DecompressionMethods.None,
+                UseCookies = false,
+                ActivityHeadersPropagator = new ReverseProxyPropagator(DistributedContextPropagator.Current)
+            });
+
+            return _defaultTransport;
+        }
+    }
+
+    /// <summary>
+    /// Adds direct forwarding of HTTP requests that match the specified pattern to a specific destination using default configuration for the outgoing request, default transforms, and default HTTP client.
+    /// </summary>
+    public static IEndpointConventionBuilder MapForwarder(this IEndpointRouteBuilder endpoints, string pattern, string serviceUrl)
+    {
+        return endpoints.MapForwarder(pattern, serviceUrl, ForwarderRequestConfig.Empty, HttpTransformer.Default, DefaultTransport);
+    }
+
+    /// <summary>
+    /// Adds direct forwarding of HTTP requests that match the specified pattern to a specific destination using customized configuration for the outgoing request, default transforms, and default HTTP client.
+    /// </summary>
+    public static IEndpointConventionBuilder MapForwarder(this IEndpointRouteBuilder endpoints, string pattern, string serviceUrl, ForwarderRequestConfig requestConfig)
+    {
+        return endpoints.MapForwarder(pattern, serviceUrl, requestConfig, HttpTransformer.Default, DefaultTransport);
+    }
+
+    /// <summary>
+    /// Adds direct forwarding of HTTP requests that match the specified pattern to a specific destination using customized configuration for the outgoing request, customized transforms, and default HTTP client.
+    /// </summary>
+    public static IEndpointConventionBuilder MapForwarder(this IEndpointRouteBuilder endpoints, string pattern, string serviceUrl, ForwarderRequestConfig requestConfig, HttpTransformer transforms)
+    {
+        return endpoints.MapForwarder(pattern, serviceUrl, requestConfig, transforms, DefaultTransport);
+    }
+
+    /// <summary>
+    /// Adds direct forwarding of HTTP requests that match the specified pattern to a specific destination using customized configuration for the outgoing request, customized transforms, and customized HTTP client.
+    /// </summary>
+    public static IEndpointConventionBuilder MapForwarder(this IEndpointRouteBuilder endpoints, string pattern, string serviceUrl, ForwarderRequestConfig requestConfig, HttpTransformer transforms, HttpMessageInvoker transport)
```

Comment:
```suggestion
    public static IEndpointConventionBuilder MapForwarder(this IEndpointRouteBuilder endpoints, string pattern, string destinationPrefix, ForwarderRequestConfig requestConfig, HttpTransformer transformer, HttpMessageInvoker httpClient)
```

These should match the naming from https://github.com/microsoft/reverse-proxy/blob/main/src/ReverseProxy/Forwarder/IHttpForwarder.cs

Reply:
My bad 😁